crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2896,11 +2896,11 @@ impl Room {
     ///
     /// This communicates to the homeserver that it should forget the room.
     ///
-    /// Only left rooms can be forgotten.
+    /// Only left or banned-from rooms can be forgotten.
     pub async fn forget(&self) -> Result<()> {
         let state = self.state();
-        if state != RoomState::Left {
-            return Err(Error::WrongRoomState(WrongRoomState::new("Left", state)));
+        if matches!(state, RoomState::Left | RoomState::Banned).not() {
```

Comment:
Can you use an explicit `match` here? Next time we add a `RoomState` variant, the compiler will complain that we need to handle it, which is nicer than having to remember to handle it ourselves.